File: Nightshade.cpp
#include "include/Nightshade.h"

Nightshade::Nightshade(World* world, const Point& position)
    : Plant(world, 99, position, "🫐", 0) {
}

Nightshade::Nightshade(World* world, const Point& position, int age)
    : Plant(world, 99, position, "🫐", age) {
}

Nightshade::~Nightshade() {

}

int Nightshade::collision(Organism& other) {
    if(this->getStrength() > other.getStrength()) {
        std::string message = "Nightshade was eaten by " + other.getSymbol();
        world->addShoutSummaryMessage(message);
        world->remove(other.getPosition());
        world->remove(this->getPosition());
        return 0;
    }
    return 0;
}

File: game-of-life-java/src/Nightshade.java
import javax.swing.*;
import java.awt.*;

public class Nightshade extends Plant {
    private static final ImageIcon nightshadeIcon = new ImageIcon("images/nightshade.png");
    public static final Image scaledNightshadeIcon = nightshadeIcon.getImage().getScaledInstance(30, 30, Image.SCALE_SMOOTH);

    public Nightshade(Point position, World world, int age) {
        super(99, position, world, age);
    }

    @Override
    public ImageIcon getIcon() {
        return nightshadeIcon;
    }

    @Override
    public void collision(Organism other) {
        String message = "Nightshade was eaten by " + other.getClass().getSimpleName();
        world.getWindow().addMessage(message);
        world.removeOrganism(this);
        world.removeOrganism(other);
    }
}
